Pod/Classes/OAStackViewDistributionStrategy.m

```diff
@@ -203,8 +212,107 @@ - (void)addProportionalityConstraintsBetween:(UIView *)view otherView:(UIView *)
 
 @end
 
+@interface OAStackViewDistributionStrategyEqualSpacing ()
+
+@property (nonatomic, strong) NSMutableArray *equalSpacingLayoutGuides;
+
+@end
+
 @implementation OAStackViewDistributionStrategyEqualSpacing
 
+- (NSLayoutAttribute)spanningAttributeForAxis:(UILayoutConstraintAxis)axis
+                                isInitialEdge:(BOOL)isInitialConstraint
+{
+  switch (axis) {
+    case UILayoutConstraintAxisHorizontal:
+      return isInitialConstraint ? NSLayoutAttributeLeading : NSLayoutAttributeTrailing;
+
+    case UILayoutConstraintAxisVertical:
+      return isInitialConstraint ? NSLayoutAttributeTop : NSLayoutAttributeBottom;
+  }
+}
+
+- (NSMutableArray *)equalSpacingLayoutGuides
+{
+  if (!_equalSpacingLayoutGuides) {
+    _equalSpacingLayoutGuides = [NSMutableArray array];
+  }
+
+  return _equalSpacingLayoutGuides;
+}
+
+- (NSString *)symbolicSpacingRelation
+{
+  return @">=";
+}
+
+- (void)alignMiddleView:(UIView *)view afterView:(UIView *)previousView
+{
+  [super alignMiddleView:view afterView:previousView];
+
+  _OALayoutGuide *guide = [_OALayoutGuide new];
+  [self.equalSpacingLayoutGuides addObject:guide];
+  [self.stackView addSubview:guide];
+
+  NSMutableArray *newConstraints = [NSMutableArray array];
+
+  UILayoutConstraintAxis axis = self.stackView.axis;
+
+  NSLayoutConstraint *firstEdgeConstraint =
+  [NSLayoutConstraint constraintWithItem:guide
+                               attribute:[self spanningAttributeForAxis:axis
+                                                          isInitialEdge:YES]
+                               relatedBy:NSLayoutRelationEqual
+                                  toItem:previousView
+                               attribute:[self spanningAttributeForAxis:axis
+                                                          isInitialEdge:NO]
+                              multiplier:1
+                                constant:0];
+
+  NSLayoutConstraint *secondEdgeConstraint =
+  [NSLayoutConstraint constraintWithItem:view
+                               attribute:[self spanningAttributeForAxis:axis
+                                                          isInitialEdge:YES]
+                               relatedBy:NSLayoutRelationEqual
+                                  toItem:guide
+                               attribute:[self spanningAttributeForAxis:axis
+                                                          isInitialEdge:NO]
+                              multiplier:1
+                                constant:0];
+
+  [newConstraints addObjectsFromArray:@[firstEdgeConstraint, secondEdgeConstraint]];
+
+  id firstGuide = self.equalSpacingLayoutGuides.firstObject;
+  if (firstGuide) {
```

Comment:
This check will always be true since you are adding the item to the array on line 254.
I think this should test if firstGuid != guid, or this will add an unneeded equality constraint were the first item and the second item is the same item.